processor/snapshotprocessor/README.md

```diff
@@ -0,0 +1,57 @@
+# Snapshot Processor
+
+The snapshot processor is used in custom distributions of the collector to provide snapshot functionality in BindPlane. It is not currently included in the official `bindplane-agent`.
+## Supported pipelines
+
+- Logs
+- Metrics
+- Traces
+
+## How it works
+
+1. The user configured the processor in the pipeline.
+2. Whenever telemetry passes through the pipeline, it is copied and stored temporarily in an internal buffer.
+3. An OpAMP server is able to use a custom message to request the contents of the internal buffer, in order to view a snapshot of the telemetry flowing through the collector.
+
+## Configuration
+
+| Field   | Type   | Default | Required | Description                                                            |
+|---------|--------|---------|----------|------------------------------------------------------------------------|
+| enabled | bool   | `true`  | `false`  | Whether the snapshot processor is enabled or not.                      |
```

Comment:
What is the purpose of it being disabled instead of just omitted from the config?

Reply:
There really isn't a purpose, I've kept it this way to be 100% compatible with the config of the previous processor.